### PR TITLE
 Add EnumAsInfo and EnumAsStateSet type overrides.

### PR DESCRIPTION
**Pull request by brian-brazil**
@SuperQ

### CODE REVIEW

**Comment by SuperQ** [comment]
Nice.

In tree.go, maybe we should only populate the final metric `EnumValues` if `Type` is not "gauge". This way it avoid filling the snmp.yml with enums that don't get used.

File: generator/tree.go [hunk 1]
```diff
@@ -322,12 +324,13 @@ func generateConfigModule(cfg *ModuleConfig, node *Node, nameToNode map[string]*
...
-				Help:    n.Description + " - " + n.Oid,
-				Indexes: []*config.Index{},
-				Lookups: []*config.Lookup{},
+				Name:       sanitizeLabelName(n.Label),
+				Oid:        n.Oid,
+				Type:       t,
+				Help:       n.Description + " - " + n.Oid,
+				Indexes:    []*config.Index{},
+				Lookups:    []*config.Lookup{},
+				EnumValues: n.EnumValues,
```
Comment (SuperQ) [inline generator/tree.go]:
```go
e := make(map[int]string)
if t != "gauge" {
  e = n.EnumValues
}
...
  EnumValues: e
...
```

**Comment by brian-brazil** [comment]
> In tree.go, maybe we should only populate the final metric EnumValues if Type is not "gauge". This way it avoid filling the snmp.yml with enums that don't get used.

I actually did that on purpose, I figure we may want to e.g. put it in the HELP for enums or something in the future and this avoids having to do a new generator run.

**Comment by SuperQ** [comment]
Interesting idea. Could possibly be huge, like ifType is.

**Comment by brian-brazil** [comment]
> Interesting idea. Could possibly be huge, like ifType is.

Yeah, I also had that concern. That's the only that big though.

**Pull request merged**